jwst/datamodels/container.py

```diff
@@ -381,6 +381,50 @@ def close(self):
             for model in self._models:
                 model.close()
 
+    @property
+    def crds_observatory(self):
+        """
+        Get the CRDS observatory for this container.  Used when selecting
+        step/pipeline parameter files when the container is a pipeline input.
+
+        Returns
+        -------
+        str
+        """
+        # Eventually ModelContainer will also be used for Roman, but this
+        # will work for now:
+        return "jwst"
+
+    def get_crds_parameters(self):
+        """
+        Get CRDS parameters for this container.  Used when selecting
+        step/pipeline parameter files when the container is a pipeline input.
+
+        Returns
+        -------
+        dict
+        """
+        with self._open_first_science_exposure() as model:
+            return model.get_crds_parameters()
+
+    def _open_first_science_exposure(self):
+        """
+        Open first model with exptype SCIENCE, or the first model
+        if none exists.
+
+        Returns
+        -------
+        stdatamodels.DataModel
+        """
+        for exposure in self.meta.asn_table.products[0].members:
```

Comment:
This `meta.asn_table` is not always guaranteed to be populated since one can use a container without an association.  So for robustness, this might have to be in a try/except.  The alternative way, if there's no association info on what is "SCIENCE" or not, is to just open the first item in the container:

```python
        return datamodel_open(self[0])
```

Reply:
@stscirij I think this is your code that I copied over from crds_client.py, do you have an opinion on this?